src/main/java/it/pagopa/interop/probing/eservice/operations/util/OffsetLimitPageable.java

```diff
@@ -34,27 +46,33 @@ public Sort getSort() {
 
 	@Override
 	public Pageable next() {
-		return null;
+		return new OffsetLimitPageable((int) getOffset() + getPageSize(), getPageSize(), getSort());
+	}
+
+	public OffsetLimitPageable previous() {
+		return hasPrevious() ? new OffsetLimitPageable((int) getOffset() - getPageSize(), getPageSize(), getSort())
+				: this;
 	}
 
 	@Override
 	public Pageable previousOrFirst() {
-		return null;
+		return hasPrevious() ? previous() : first();
 	}
 
 	@Override
 	public Pageable first() {
-		return null;
+		return this;
```

Comment:
shouldn't this be
```suggestion
		return return new OffsetLimitPageable(0, getPageSize(), getSort());
```
?

Reply:
done in the subsequent commit